src/main/kotlin/org/amshove/kluent/Exceptions.kt

```diff
@@ -9,9 +9,11 @@ infix fun <T : Throwable> (() -> Any?).shouldThrow(expectedException: KClass<T>)
         fail("There was an Exception expected to be thrown, but nothing was thrown", "$expectedException", "None")
     } catch (e: Throwable) {
         @Suppress("UNCHECKED_CAST")
-        if (e.isA(ComparisonFailure::class)) throw e
-        else if (e.isA(expectedException)) return ExceptionResult(e as T)
-        else throw ComparisonFailure("Expected ${expectedException.javaObjectType} to be thrown", "${expectedException.javaObjectType}", "${e.javaClass}")
+        return when {
+            e.isA(AssertionError::class) -> throw e
+            e.isA(expectedException) -> ExceptionResult(e as T)
+            else -> throw AssertionFailedError("Expected ${expectedException.javaObjectType} to be thrown", "${expectedException.javaObjectType}", "${e.javaClass}")
```

Comment:
Can this be replaced with `fail()` from `kotlin-test` to get rid of the opentest4j dependency?

Reply:
Is there a reason for removing `opentest4j`? As [I said earlier](https://github.com/MarkusAmshove/Kluent/pull/72#issuecomment-340747784), it's Java 6 compliant and *the industry standard* for the exception to be thrown by assertion libraries.

`kotlin-test`'s `fail` uses `AssertionError` which does not express the errors as richly.

Reply:
Yeah, agree with the reasoning. Indeed, what I was driving to is reducing the dependency on Java classes and libraries, but as you rightly mentioned, this has to be a separate discussion. I'd be fine with opentest4j for now.